src/frontends/onnx/frontend/src/core/graph.cpp

```diff
@@ -384,10 +384,25 @@ OutputVector Graph::make_ng_nodes(const Node& onnx_node) {
         std::rethrow_exception(std::current_exception());
     }
 
+    size_t output_size = 0;
+    std::for_each(std::begin(ng_subgraph_outputs),
+                  std::end(ng_subgraph_outputs),
+                  [&output_size](const Output<ov::Node>& output) {
+                      output_size += output.get_node()->get_output_size();
+                  });
+    NGRAPH_CHECK(onnx_node.get_outputs_size() == output_size,
```

Comment:
yes, it was a case from the custom op registration (where an user has very unclear error message)

Reply:
I mean, for example `ONNX NonMaxSuppression` has only `1` output while `OpenVINO NonMaxSuppression` has `3 outputs`.

Also please take a look at the failing tests: https://dev.azure.com/openvinoci/dldt/_build/results?buildId=436813&view=logs&j=2f2823e7-545a-5ed6-9d73-63b6cd69c54b&t=ffe810cc-1cea-5574-ed02-77983a59b25d&l=5220.

Seems that `onnx_node.get_outputs_size() == output_size` requirement breaks backward compatibility.

Reply:
Maybe it can be relaxed to: `onnx_node.get_outputs_size() <= ng_subgraph_outputs.size()`
```suggestion
    NGRAPH_CHECK(onnx_node.get_outputs_size() <= ng_subgraph_outputs.size(),
```

Reply:
@mitruska yes, it makes sense for such cases, thank you ;)